lib/scalyr/common/util.rb

```diff
@@ -45,5 +45,25 @@ def self.flatten(obj, delimiter='_')
   return result
 end
 
+
+
+def self.calculate_percentiles(values, percentiles)
```

Comment:
Hm, if I'm reading metrics recording code correctly, there is no upper bound non the ``values`` array size?

If that is indeed the case, we need to modify this a bit to have an upper bound to ensure this array doesn't grow larger than some max count.

In ideal world, we could report raw or sampled values to the server side more often and then handle percentile calculation on the server side, but that's likely not feasible at the moment.

Simplest option would be to use uniform sampling. Here is how one of the Python libraries implements it - https://github.com/avalente/appmetrics/blob/master/appmetrics/histogram.py.

EDIT: It looks like this may work - https://github.com/matttproud/ruby_quantile_estimation. It uses estimator over data stream with a maximum limit of observed values it keeps in memory (512). It won't be 100% accurate, but that's enough for our use case.